modules/ROOT/pages/cross_cutting/oauth2_proxy.adoc

```diff
@@ -0,0 +1,70 @@
+:imagesdir: ../images
+= OAuth2 Proxy
+
+link:https://OAuth2-proxy.github.io/OAuth2-proxy/[OAuth2 proxy] is a link:https://en.wikipedia.org/wiki/Reverse_proxy[reverse proxy] that runs in front of your services to handle the complexity of link:https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-OAuth-and-oidc[OpenID Connect / OAuth 2].
+Applications behind the proxy can be sure, that requests to them have already been authorized. 
+The proxy supports a variety of different link:https://en.wikipedia.org/wiki/Identity_provider[identity providers] like Google, GitHub, Keycloak...
+
+== Authentication flow
+
+A detailed flow diagramm can be found link:https://github.com/oauth2-proxy/oauth2-proxy/issues/1438[here]
+
+== Setup
+
+The OAuth proxy can be deployed from a link:https://github.com/OAuth2-proxy/OAuth2-proxy/releases/latest[prebuilt binary], a prebuilt link:https://quay.io/repository/OAuth2-proxy/OAuth2-proxy?tab=tags&tag=latest[docker image] or by link:https://github.com/OAuth2-proxy/manifests[helm chart] inside a Kubernetes cluster. It's recommended to use Kubernetes.
+
+image::OAuth2_Proxy-RequestFlow.drawio.svg[Request Flow]
+
+
+=== Keycloak 
+
+In this example, link:https://www.keycloak.org/[keycloak] is used as identity provider. For more details on keycloak see the devon link:https://devonfw.com/docs/java/current/cross_cutting/keycloak.html[documentation]. 
+To deploy keycloak in the cluster, it's recommended to use the helm chart from link:https://artifacthub.io/packages/helm/bitnami/keycloak[bitnami]. To configure the keycloak deployment, use a link:https://github.com/bitnami/charts/blob/main/bitnami/keycloak/values.yaml[values.yaml] file. After keycloak is running go to the keycloak admin ui and configure a client. Make sure to follow the link:https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider#keycloak-oidc-auth-provider[guide of OAuth2] proxy to set up the group mapping properly.
+
+----
+helm repo add bitnami https://charts.bitnami.com/bitnami
+
+helm upgrade --install keycloak bitnami/keycloak -f config/keyCloakValues.yaml
+----
+
+=== Keycloak ingress
+
+This ingress is created by the keycloak helm chart, allowing a host like `keycloak.localtest.me` to be mapped to the keycloak service. This is important for setting up redirect URLs in the OAuth2 proxy.
```

Comment:
What is localtest.me. Why do I need it ;-) 

I see two options here:
1. Make clear that it's all about a local setup and explain the localtest.me aso.
2. Make it more abstract. "A ingress gateway is necessary to access keycload" -> Leading me to the question "Is really a ingress necessary? Or would a load balancer service be enough? If an ingress is necessary why?
It's important to make clear that we're now

Reply:
I think option two would be best and most helpful. Less details more conceptual description

Reply:
The keycloak ingress can be  replaced with an extra entry in the main ingress. I will remove the keycloak ingress and update the documentation.